### PR TITLE
`auth` Allow override of AWS Resolver URL

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
- Allows users to override AWS resolver URL for identities and providers of AWS cloud.

## why
- Allows us to override the resolver URL. This is useful in cases where you need to stub out the AWS endpoint.
- This will be useful for testing and for usage with localstack.

## Summary by CodeRabbit

- **New Features**
  - Added support for a configurable custom AWS endpoint resolver used by AWS authentication flows; can be configured at identity or provider level (identity takes precedence).

- **Documentation**
  - Added guidance and YAML examples for configuring custom AWS endpoints, including a LocalStack usage example.

- **Tests**
  - Added unit tests covering resolver precedence, empty/nil configs, extraction logic, and edge cases.

- **Examples**
  - Demo updated with a default LocalStack superuser identity for local testing.

### CODE REVIEW

File: pkg/auth/providers/aws/sso_test.go [hunk 1]
```diff
@@ -204,3 +204,50 @@ func TestSSOProvider_promptDeviceAuth_NonCI_OpensURL(t *testing.T) {
...
+		StartURL: "https://company.awsapps.com/start",
+	}
+
+	p, err := NewSSOProvider("sso-standard", config)
+	require.NoError(t, err)
+	assert.NotNil(t, p)
+	assert.Equal(t, "sso-standard", p.Name())
+
+	// Verify the provider works without resolver config
+	assert.NoError(t, p.Validate())
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/auth/providers/aws/sso_test.go]: @Benbentwo make sure you have precommit hooks enabled.
